### ISSUE TITLE
Trouble using flycheck-add-next-checker

### ISSUE THREAD

**Issue by fmichaut-dev** [flycheck/flycheck#1928]
This template is for **user support**.  If you are reporting an actual Flycheck bug, please use the relevant issue template instead.

## Checklist
- [x] I have searched existing issues and StackOverflow for solutions.
- [x] I have read the [Flycheck manual][].
- [x] I have read the [Troubleshooting Guide][].
- [x] I have checked my issue has not been reported yet.

## Problem description
I can't figure out a clean way to have `clang-tidy` checker run after `irony`.
Either `flycheck-add-next-checker` gives me `irony is not a valid syntax checker` error, or `clang-tidy` ends up beeing the first checker to run and totally bypass irony (and I couldn't find a way to define which checher should run first when 2 are candidates in a buffer).

My current config is that, but it looks overcomplicated : 
```
(eval-after-load 'flycheck
    '(add-to-list 'flycheck-checkers 'c/c++-clang-tidy))
(eval-after-load 'flycheck
  '(add-hook 'flycheck-mode-hook (lambda () (flycheck-add-next-checker 'irony '(warning . c/c++-clang-tidy)))))
(eval-after-load 'flycheck
  '(add-hook 'flycheck-mode-hook #'flycheck-irony-setup))
(eval-after-load 'flycheck
  '(add-hook 'flycheck-mode-hook #'flycheck-clang-tidy-setup))
```

## System configuration
```
Syntax checkers for buffer main.cpp in c++-mode:                                                                                                                                              

First checker to run:

  irony
    - may enable:    yes
    - may run:       t
    - Irony Mode:    enabled
    - irony-server:  Found at /home/fmichaut/.emacs.d/.local/etc/irony-server/bin/irony-server
    - next checkers: c/c++-clang-tidy, clang-analyzer

Checkers that may run as part of the first checker's chain:

  c/c++-clang-tidy
    - may enable:         yes
    - may run:            t
    - executable:         Found at /usr/bin/clang-tidy
    - configuration file: Found at "/home/fmichaut/.clang-tidy"
    - next checkers:      c/c++-cppcheck

  clang-analyzer
    - may enable: yes
    - may run:    t
    - executable: Found at /usr/bin/clang
    - Backend:    irony

Checkers that could run if selected:

  c/c++-clang  select
    - may enable:    yes
    - executable:    Found at /usr/bin/clang
    - next checkers: clang-analyzer, c/c++-cppcheck

  c/c++-gcc  select
    - may enable:    yes
    - executable:    Found at /usr/bin/gcc
    - next checkers: c/c++-cppcheck

Checkers that are compatible with this mode, but will not run until properly configured:

  c/c++-cppcheck (automatically disabled) reset
    - may enable: no
    - executable: Not found


Flycheck Mode is enabled. Use C-u C-c ! x to enable disabled checkers.

Flycheck version: 32snapshot
Emacs version:    27.2
System:           x86_64-redhat-linux-gnu
Window system:    nil 
```

Emacs configuration:
- [ ] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [x] Doom Emacs
- [ ] Other shared configuration

## Additional notes
I opened a similar issue on `flycheck-irony` : https://github.com/Sarcasm/flycheck-irony/issues/26

[Flycheck manual]:  https://www.flycheck.org/en/latest/user/troubleshooting.html
[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html
